### PR TITLE
Translate all the things

### PR DESCRIPTION
**Pull request by tdroxler**
Resolves: #207
I think I translate almost everything, I still need to do the date/hour with `dayjs`

### CODE REVIEW

File: locales/fr-FR/translation.json [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  "Search for an address or a tx...": "Rechercher une adresse ou une transaction...",
+  "Search": "Rechercher",
+  "See more": "Voir plus",
+  "Self transfer": "Auto-transfert",
+  "Status": "Status",
+  "Success": "Succès",
+  "Supply": "Emission",
+  "Timestamp": "Date",
+  "Tokens": "Jetons",
+  "Total Amount": "Montant Totale",
```
Comment (mvaivre) [inline locales/fr-FR/translation.json:60]: Montant Total

File: src/components/LanguageSwitch.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+  const items = languageOptions.map((lang) => ({
+    text: lang.label,
+    onClick: () => setLangValue(lang.value)
+  }))
+
+  return <Menu aria-label="Lang" label={langValue} items={items} direction="up" className={className} />
+}
+
+export default styled(LanguageSwitch)``
```
Comment (mvaivre) [inline src/components/LanguageSwitch.tsx:61]: No need to wrap this as a styled component (`styled`) if you don't apply any style

File: src/pages/BlockInfoPage.tsx [hunk 3]
```diff
@@ -160,15 +162,15 @@ const BlockInfoPage = () => {
...
+          <InlineErrorMessage message={t('An error occured while fetching transactions')} code={txListStatus} />
         ) : (
           <>
             <SecondaryTitle>Transactions</SecondaryTitle>
             <Table main hasDetails scrollable isLoading={txLoading}>
               {txList && txList && (
                 <>
                   <TableHeader
-                    headerTitles={['', 'Hash', 'Inputs', '', 'Outputs', 'Total Amount', '']}
+                    headerTitles={['', t('Hash'), t('Inputs'), '', t('Outputs'), t('Total Amount'), '']}
```
Comment (mvaivre) [inline src/pages/BlockInfoPage.tsx:173]: Idea: we could modify the `TableHeader` component to apply the translation function there (in order to just pass strings here). WDYT?

**Review by mvaivre — COMMENTED** [review]
Great job! 👏 
Just a few comments.

I'm yet to fo the functional testing.

File: src/hooks/useTransactionUI.ts [hunk 4]
```diff
@@ -33,20 +34,21 @@ interface TransactionUIProps {
 // TODO: Better (better way to define infoType by looking at presence of script)
 // Use script field in tx once available https://github.com/alephium/explorer-backend/issues/485
 
-export const getTransactionUI = ({
+export const GetTransactionUI = ({
```
[the remark this answers is not in the document]
Comment (mvaivre) [reply src/hooks/useTransactionUI.ts:37]: You made the compiler think it's a component when it's now.

We need to import the i18n object here.

[the remark this answers is not in the document]
Comment (tdroxler) [reply src/hooks/useTransactionUI.ts]: done

File: locales/fr-FR/translation.json (hunk 1 above)
Comment (tdroxler) [reply locales/fr-FR/translation.json]: done

File: src/components/LanguageSwitch.tsx (hunk 2 above)
Comment (tdroxler) [reply src/components/LanguageSwitch.tsx]: done

File: src/pages/BlockInfoPage.tsx (hunk 3 above)
Comment (tdroxler) [reply src/pages/BlockInfoPage.tsx]: done :rocket: much nicer

**Comment by mvaivre** [comment]
I refined a couple of things, good to go now, thanks @tdroxler !

**Review by mvaivre — APPROVED** [review]

**Pull request merged**